src/_pytest/nodes.py

```diff
@@ -135,18 +137,48 @@ def __repr__(self):
         return "<%s %r>" % (self.__class__.__name__, getattr(self, "name", None))
 
     def warn(self, code, message):
```

Comment:
i just had a really interesting idea - lets add a parameter called code_or_warning at the beginning and make message optional

then we can do `node.warn(Awarning(...))` and jsut deprecate the old invocation

Reply:
Ahh good idea, will do.

Reply:
Done!